### ISSUE TITLE
Implement Asynchronous Processing 

### ISSUE THREAD

**Issue by plunkettscott** [laravel-json-api/laravel#10]
Implement Asynchronous Processing as described in the [JSON:API Spec recommendation](https://jsonapi.org/recommendations/#asynchronous-processing). This exists in the outgoing version of this package but has not been implemented in this version.

**Comment by lindyhopchris** [comment]
Thanks for opening the issue!

**Comment by plunkettscott** [comment]
No problem. If I get some time I can take a look at implementing this as well. Any ideas how you'd like to see something like this implemented?

**Comment by lindyhopchris** [comment]
Thanks for the offer, it is appreciated but I think I need to do this one. I want it to be in a separate package: `laravel-json-api/async` as it's an extra feature rather than a core feature.

**Comment by lindyhopchris** [comment]
Note to self: there was an issue on the old package about investigating improving the implementation by using Job Middleware:
https://github.com/cloudcreativity/laravel-json-api/issues/418#issue-490940922

**Comment by zlodes** [comment]
@lindyhopchris Hi! Did you have any plans about it? 
It's just the last thing that don't let us to upgrade to the brand-new package.

P.S.: It's amazing and much better than the previous one. Good job! Thank you!

**Comment by lindyhopchris** [comment]
@zlodes thanks for the feedback, it's really good to hear that it's an improvement - I also prefer the new package!

The only thing blocking this is time at the moment. The production APIs I'm working on at the moment don't use the async feature, which is why it hasn't bubbled up to the top of my priorities. It's really difficult for me to estimate at the moment when I might get on to it.

**Comment by makowskid** [comment]
Hi! Any update on this topic? Any plans to actually add async processing?

**Comment by lindyhopchris** [comment]
Yeah it's on the list of things I need to work through for #238 - I made progress on that ticket during the summer but have had to put it down again due to other priorities. Planning to pick back up sometime soon.
